src/coreclr/scripts/superpmi.py

```diff
@@ -244,7 +244,8 @@
 collect_parser.add_argument("--pmi", action="store_true", help="Run PMI on a set of directories or assemblies.")
 collect_parser.add_argument("--crossgen", action="store_true", help="Run crossgen on a set of directories or assemblies.")
 collect_parser.add_argument("--crossgen2", action="store_true", help="Run crossgen2 on a set of directories or assemblies.")
-collect_parser.add_argument("-assemblies", dest="assemblies", nargs="+", default=[], help="Pass a sequence of managed dlls or directories to recursively use while collecting with PMI, crossgen, or crossgen2. Required if --pmi, --crossgen, or --crossgen2 is specified.")
+collect_parser.add_argument("-assemblies", dest="assemblies", nargs="+", default=[], help="A list of managed dlls or directories to recursively use while collecting with PMI, crossgen, or crossgen2. Required if --pmi, --crossgen, or --crossgen2 is specified.")
+collect_parser.add_argument("-exclude", dest="exclude", nargs="+", default=[], help="A list of files or directories to exclude from the files and directories specified by `-assemblies`.")
```

Comment:
I would recommend the right way to do this would be to not copy the assemblies that are problematic to the `payload` directory. `superpmi-setup.py` already has an ability to do that:
https://github.com/dotnet/runtime/blob/4de28f317c227b433fafecd5496bae04c2bca007/src/coreclr/scripts/superpmi-setup.py#L296

Perhaps, just add a condition there that if the `collection_name == 'crossgen2'`, exclude it.

Reply:
With that, you won't need any changes in `superpmi.py`. The script will just act on the `assemblies` you provided.

Reply:
ok, I moved the exclusion to superpmi-setup.py. The superpmi.py `-exclude` code is not required now, but I'd like to keep it as it could be generally useful.